pkg/controller/extensions/extension/extension.go

```diff
@@ -241,6 +246,41 @@ func (h *extensionHandler) processDeployment(ctx context.Context) error {
 	return nil
 }
 
+func (h *extensionHandler) processJob(ctx context.Context) error {
+	controllerJob := h.ext.Spec.Controller.Job
+	if controllerJob == nil {
+		return nil
+	}
+
+	// ensure the job is set to use this extension's service account that we created
+	jobSpec := *controllerJob.Spec.DeepCopy()
+	jobSpec.Template.Spec.ServiceAccountName = h.ext.Name
+
+	ref := meta.AsOwner(meta.ReferenceTo(h.ext, v1alpha1.ExtensionGroupVersionKind))
+	j := &batch.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            controllerJob.Name,
+			Namespace:       h.ext.Namespace,
+			OwnerReferences: []metav1.OwnerReference{ref},
+		},
+		Spec: jobSpec,
+	}
+	if err := h.kube.Create(ctx, j); err != nil && !kerrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, "failed to create job")
+	}
+
+	// save a reference to the extension's controller
+	h.ext.Status.ControllerRef = &corev1.ObjectReference{
```

Comment:
You could also use `meta.ReferenceTo(j, JobGroupVersionKind)` here if you wanted.

Reply:
Is 60dc62f0292f3f6ead6170c81cebed1224164eda what you were thinking?

Reply:
Does `d.GroupVersionKind()` always return the GVK of a deployment? I'm unfamiliar with that particular method, but I know it's quite common for objects to attempt to return their GVK based on their embedded type metadata, which is often unset.

Reply:
Good catch.  `TypeMeta` is not explicitly set here,
https://github.com/crossplaneio/crossplane/pull/583/files#diff-0dc4437d46271355bb75e3c9b14295ddR225-R232 .. 
So what response does this cause ellicit from `GroupVersionKind()`? .. `schema.GroupVersionKind{"","",""}`

(I took advantage of #579 to dig into this)

I'm defining the GVK in advance in d7e8d08 and referencing that when setting the controller reference.

Reply:
You could also lookup the Job or Deployment's GVK in the controller manager's scheme, roughly like:

```go
type extensionHandler struct {
	kube  client.Client
        typer runtime.ObjectTyper  // Satisfied by mgr.GetScheme()
	ext   *v1alpha1.Extension
}

// Assume eh is an &extensionHandler{}
ref := meta.ReferenceTo(j, resource.MustGetKind(j, eh.typer)
```

Reply:
I'm having a hard time figuring out the least obtrusive way to fit that mgr object into the extensionHandlerFactory `newHandler` method.

What is the benefit of using the `MustGetKind(...typer)` approach compared to using hard coded GKV (especially considering that this follows the creation of the Kubernetes object using typed objects -- `&appv1.Deployment{}`)?

Reply:
> What is the benefit of using the MustGetKind(_, typer) approach compared to using hard coded GKV

I don't think there's an obvious benefit, so if it's hard to do a hard coded GVK should be fine.